vscode-lean4/media/manual.md

```diff
@@ -0,0 +1,631 @@
+
+| &nbsp; |
+| :--: | 
+| ![](media/diagnostics.png) | 
+| *Error, warning and information diagnostics with corresponding hover- and InfoView information. The red / orange / blue areas in the scroll bar denote the location of the area. The bottom left of the status bar at the bottom of VS Code displays the amount of diagnostics in all open files and clicking this section of the status bar will open the 'Problems view'.* |
+| &nbsp; |
+| ![](media/problems-view.png) | 
+| *VS Code 'Problems view'* |
+
+### File restarting
+
+When a file `Module.lean` imports another file `Submodule.lean`, changes in `Submodule.lean` do not automatically become visible in `Module.lean`. This is to prevent accidentally triggering expensive builds of all dependents of `Submodule.lean`.
+
+Instead, the state of all dependencies of `Module.lean` needs to be updated manually using the ['Server: Restart File'](command:lean4.restartFile) command. This will trigger a build for all changed dependencies. The command can be called using `Ctrl+Shift+X` (`Cmd+Shift+X`), using the 'Restart File' button in the InfoView, using the [command menu](#command-menu), using the [command palette](#command-palette) or by right clicking in the editor of a Lean file and selecting the 'Server: Restart File' entry in the context menu.
+
+VS Code will display [error- or information-level diagnostics](#errors-warnings-and-information) whenever a dependency of a file is edited and saved. When initially opening a file and a dependency would need to be rebuilt to be up-to-date, it issues an error and refuses to [process](#file-processing) the rest of the file. When a dependency changes while the file is already open, it issues an information-level diagnostic, but will continue processing the file with the previous state of the dependency.
+
+In order to automatically trigger rebuilds of all changed dependencies when opening a file instead of issuing an error, the 'Lean 4: Automatically Build Dependency' setting can be enabled.
+
+| &nbsp; |
+| :--: | 
+| ![](media/restart-file-infoview-button.png) | 
+| *'Restart File' button in InfoView* |
+
+### Unicode input
+
+Lean code uses lots of [Unicode symbols](https://home.unicode.org/) to aid readability. In Lean files, these Unicode symbols can be entered by typing a backslash followed by an abbreviation identifier referring to the symbol. As soon as the abbreviation identifier is unique, the abbreviation will be automatically replaced with the corresponding Unicode symbol - for example, `\forall` will yield `∀`. 
+
+To replace an abbreviation early, before it is unique, `Tab` can be pressed to trigger the ['Lean 4: Input: Convert Current Abbreviation'](command:lean4.input.convert) command. This will yield the Unicode symbol with the shortest abbreviation identifier.
```

Comment:
I think it should also be mentioned that space and enter (and also right-arrow, and maybe some other keys I don't know about) also cause the identifier to be replaced.